std/exception.d

```diff
@@ -386,12 +385,17 @@ void assertThrown(T : Throwable = Exception, E)
         If a delegate is passed, the safety and purity of this function are inferred
         from `Dg`'s safety and purity.
  +/
-T enforce(E : Throwable = Exception, T)(T value, lazy const(char)[] msg = null,
-string file = __FILE__, size_t line = __LINE__)
-if (is(typeof({ if (!value) {} })))
+template enforce(E : Throwable = Exception)
+if (is(typeof(new E("", __FILE__, __LINE__)) : Throwable) || is(typeof(new E(__FILE__, __LINE__)) : Throwable))
 {
-    if (!value) bailOut!E(file, line, msg);
-    return value;
+
+    T enforce(T)(T value, lazy const(char)[] msg = null,
+    string file = __FILE__, size_t line = __LINE__)
+    if (is(typeof({ if (!value) {} })))
```

Comment:
We have `ifTestable`, though I have to say this is nicer.

Reply:
Yes, `ifTestable` is one of those one-liners that cost a lot more than they are worth it:

https://github.com/dlang/phobos/blob/3f30d5c02bae9c5b2af3d3e5e2602f0f5403a547/std/traits.d#L7996

Interestingly it crept into Phobos with other one-liners: https://github.com/dlang/phobos/pull/3395